llvm/lib/Target/X86/X86InstrAVX10.td

```diff
@@ -1225,3 +1225,45 @@ defm VFNMADD132NEPBF16 : avx10_fma3p_132_bf16<0x9C, "vfnmadd132nepbf16", X86any_
 defm VFNMSUB132NEPBF16 : avx10_fma3p_132_bf16<0x9E, "vfnmsub132nepbf16", X86any_Fnmsub,
                                               X86Fnmsub, SchedWriteFMA>;
 }
+
+//-------------------------------------------------
+// AVX10  COMEF instructions
+//-------------------------------------------------
+multiclass avx10_com_ef_int<bits<8> Opc, X86VectorVTInfo _, SDNode OpNode,
+                             string OpcodeStr,
+                             Domain d,
+                             X86FoldableSchedWrite sched = WriteFComX> {
+  let ExeDomain = d in {
+    def rr_Int : AVX512<Opc, MRMSrcReg, (outs), (ins _.RC:$src1, _.RC:$src2),
+                        !strconcat(OpcodeStr, "\t{$src2, $src1|$src1, $src2}"),
+                        [(set EFLAGS, (OpNode (_.VT _.RC:$src1), _.RC:$src2))]>,
+                        EVEX, EVEX_V128, Sched<[sched]>, SIMD_EXC;
+    let mayLoad = 1 in {
+        def rm_Int : AVX512<Opc, MRMSrcMem, (outs), (ins _.RC:$src1, _.ScalarMemOp:$src2),
+                            !strconcat(OpcodeStr, "\t{$src2, $src1|$src1, $src2}"),
+                            [(set EFLAGS, (OpNode (_.VT _.RC:$src1), (_.LdFrag addr:$src2)))]>,
+                            EVEX, EVEX_V128, Sched<[sched]>, SIMD_EXC;
+    }
+  }
+}
+
+let Defs = [EFLAGS], Predicates = [HasAVX10_2] in {
+  defm VCOMXSDZ   :  avx10_com_ef_int<0x2f, v2f64x_info, X86comi512,
```

Comment:
We still need to define the SAE variants here. We just don't need define patterns for them.

Reply:
we dont have separate record for VCOMXSDZ   and VCOMXSDZ_SAE.
We used VCOMXSDZrr_Int and VCOMXSDZrrb_Int to separate out SAE version.
 definition of rrb_int used mayRaiseFPException=0 and It required SAE node pattern. SAE node pattern was deleted since we had no way to reach/exercise it from COMI Type.

Reply:
done.

Reply:
Please also add enc/dec tests.